crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -1112,6 +1115,82 @@ impl BeaconState {
 
         Ok(())
     }
+
+    pub fn process_attestation(&mut self, attestation: Attestation) -> anyhow::Result<()> {
```

Comment:
```suggestion
    pub fn process_attestation(&mut self, attestation: &Attestation) -> anyhow::Result<()> {
```

IMO as this function doesn't have to take the ownership of `attestation`, just giving reference would be feasible

Reply:
We need a mut reference for this, no?
https://github.com/varun-doshi/ream/blob/a8c2416d54571758be1588bc32fff5578f4e3893/crates/common/consensus/src/deneb/beacon_state.rs#L1191

Reply:
What I mean was Attestation doesn't have to be an actual value. Yes, BeaconState has to be mutable reference as you mentioned

Reply:
Ahh yes